### PR TITLE
Remove python2 compatibility code

### PR DESCRIPTION
**Pull request by swamper123**
Drop the Python2 and <3.6 code

### CODE REVIEW

**Comment by dhoomakethu** [comment]
@swamper123 Apologies for the long delay, looks like you have commited `venv` as well with this PR. Would be great If you could clean this up.

File: README.rst [hunk 1]
```diff
@@ -120,7 +119,7 @@ at gitter:  https://gitter.im/pymodbus_dev/Lobby
 ------------------------------------------------------------
 Pymodbus REPL (Read Evaluate Print Loop)
 ------------------------------------------------------------
-Starting with Pymodbus 2.x, pymodbus library comes with handy
+Starting with Pymodbus 3.x, pymodbus library comes with handy
```
Comment (alexrudd2) [inline README.rst:122]: This should not have been changed

**Comment by swamper123** [comment]
FYI: At the moment I am still busy with other things going on. So it will either be at the end of the year or mid January, unti I can lay my hands on that PR again.

**Comment by swamper123** [comment]
I think I got the most stuff uptodate again.
Not sure, what the failing report tries to tell me.
If something is still missing/not correct: Help is always wanted ;)

**Comment by dhoomakethu** [comment]
It could be the coverage that has dropped down.

**Comment by swamper123** [comment]
The stuff may be green, but I am still not happy about this, but don't know where to continue a.t.m.:

```bash
====================== 474 passed, 19 skipped in 42.13s =======================
D:\a\pymodbus\pymodbus\.tox\pypy37\site-packages\_pytest\mark\structures.py:552: RuntimeWarning: coroutine 'AsyncioModbusSerialClient.connect' was never awaited
  seen.update(self.parent.keywords)
D:\a\pymodbus\pymodbus\.tox\pypy37\site-packages\_pytest\mark\structures.py:552: RuntimeWarning: coroutine 'ReconnectingAsyncioModbusTcpClient._reconnect' was never awaited
  seen.update(self.parent.keywords)
```

I don't know if that happens in branch 3.0.0 as well, because the logs are to old and have been removed. 
Not sure if it is the `mock`stuff or a factory thing...

**Comment by swamper123** [comment]
Would be happy about every review. :)

**Comment by swamper123** [comment]
I know that this is a bigger PR, but a step in an easier to handle direction in the future.
Still glad about every review that would be made.

**Comment by dhoomakethu** [comment]
I am biting the bullet and merging this to 3.0.0 branch and will release `3.0.0.dev3` soon

**Pull request merged**